src/Bridge/Doctrine/MongoDbOdm/PropertyInfo/DoctrineExtractor.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\PropertyInfo;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Persistence\Mapping\MappingException;
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ODM\MongoDB\Mapping\MappingException as MongoDbOdmMappingException;
+use Doctrine\ODM\MongoDB\Types\Type as MongoDbType;
+use Symfony\Component\PropertyInfo\PropertyListExtractorInterface;
+use Symfony\Component\PropertyInfo\PropertyTypeExtractorInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Extracts data using Doctrine MongoDB ODM metadata.
+ *
+ * @experimental
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class DoctrineExtractor implements PropertyListExtractorInterface, PropertyTypeExtractorInterface
+{
+    private $objectManager;
+
+    public function __construct(ObjectManager $objectManager)
+    {
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getProperties($class, array $context = [])
+    {
+        try {
+            $metadata = $this->objectManager->getClassMetadata($class);
+        } catch (MappingException $exception) {
+            return null;
+        } catch (MongoDbOdmMappingException $exception) {
```

Comment:
This exception extends the base mapping exception, so the second `catch` will never be hit.